src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -14,7 +14,7 @@ Copyright (c) .NET Foundation. All rights reserved.
   <PropertyGroup>
     <DefaultCopyToPublishDirectoryMetadata Condition="'$(DefaultCopyToPublishDirectoryMetadata)' == ''">true</DefaultCopyToPublishDirectoryMetadata>
     <_GetChildProjectCopyToPublishDirectoryItems Condition="'$(_GetChildProjectCopyToPublishDirectoryItems)' == ''">true</_GetChildProjectCopyToPublishDirectoryItems>
-    <IsPublishable Condition="'$(IsPublishable)' == ''">true</IsPublishable>
+    <IsPublishable Condition="'$(IsPublishable)' == '' and '$(OutputType)' != 'Library'">true</IsPublishable>
```

Comment:
Without this change, Library projects participate in Publish. previously this value was assumed to be true because different project types deliberately Imported the Publish SDK. This is no longer the case, so we have to be a bit more proscriptive.